### ISSUE TITLE
Excel: NVDA does not announce when ENTER is pressed to move to a new line

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#6500]
When adding data into an Excel spreadsheet, a convenient way of moving is by pressing TAB to move to the cell to the right, then ENTER to move to the next row, at the first column you pressed TAB from, announcing the new cell as it moves.

This works fine if you enter data into cells as you go.  If you press ENTER without having typed anything into the last cell, the focus moves as expected, however, NVDA does not announce anything.  If NVDA+TAB is pressed at this point, NVDA also does not read the correct cell (it reads the previous cell).

To replicate:
1. Open a blank workbook
2. Type something in A1
3. Press ENTER.  Note that NVDA reads that the focus has moved to A2.
3. Press ENTER again.  The focus has moved to A3 however NVDA does not read anything.
4. Press NVDA+tab.  NVDA reads A2.
5. Press TAB to move to the next column.  NVDA reads B3, confirming that the focus at step 4 was actually in A3.

Immediate workaround is to recommend to use arrows to navigate if not editing cell contents, however this situation could be confusing if someone pressed ENTER, particularly if they then press NVDA+TAB to confirm where the focus is.

Tested in both Excel 2007 and 2016.

**Comment by jcsteh** [comment]
P2 because this could be very confusing and because the fix is ridiculously easy.

As a side note, I didn't realise enter worked in this situation. I assumed enter was only valid when editing. Pressing enter to move to the next row when not entering seems counter-intuitive to me, but maybe it's just me. :)

**Comment by Qchristensen** [comment]
For the most part, if just reading, then yes, I would tend to use the
arrows.  Theoretically, though, you might be filling in a spreadsheet, tab
across a line and realise you don't actually need to enter data in the last
column and press ENTER, which would trigger this bug.


-- 
22 Point

Web: http://www.22point.com.au
Facebook: https://www.facebook.com/22Point
Twitter: https://twitter.com/22PointApps

Check out our first app, RapiTap! - Tap targets fast & avoid decoys:
Adrenaline pumping, Challenging, Accessible!
https://play.google.com/store/apps/details?id=au.com.twentytwopoint.rapitap
Free trial:
https://play.google.com/store/apps/details?id=au.com.twentytwopoint.rapitapfree
